### PR TITLE
Add a sanitation to OpenAPI spec

### PR DESCRIPTION
**Pull request by SanduniU**
## Purpose
Add a sanitation to open API spec to resolve a payload binding error.

Changes
- Added the sanitation to OpenAPI specification
- Documented the sanitation in sanitations.md
- Included the generated types.bal file

### CODE REVIEW

File: docs/spec/sanitations.md [hunk 1]
```diff
@@ -83,13 +83,27 @@ This document records the sanitation done on top of the official OpenAPI specifi
...
+   - **Reason**: This change is necessary to prevent binding errors that occur when there are no messages in the thread, as `first_id` and `last_id` need to be nullable in such cases.
+
 ---
 
 ## OpenAPI cli command
 
 The following command was used to generate the Ballerina client from the OpenAPI specification. The command should be executed from the repository root directory.
 
 ```bash
-bal openapi -i docs/spec/openapi.yaml --mode client --tags Assistants --license docs/license.txt -o ballerina
```
Comment (ayeshLK) [inline docs/spec/sanitations.md]: Should this be `Assistants` or `Images` ?

Comment (SanduniU) [reply docs/spec/sanitations.md]: It should be Assistants. It is fixed now

Comment (ayeshLK) [reply docs/spec/sanitations.md]: Fixed with [8cf206d2f805874d5aae9fea8dfddec5405c8dcb](https://github.com/ballerina-platform/module-ballerinax-openai.assistants/pull/8/commits/8cf206d2f805874d5aae9fea8dfddec5405c8dcb)

**Comment by ayeshLK** [comment]
In the OpenAPI spec you have only done changes to `ListMessagesResponse` schema. But, I see several other types generated as well. Any reason for that ?

**Comment by SanduniU** [comment]
> In the OpenAPI spec you have only done changes to `ListMessagesResponse` schema. But, I see several other types generated as well. Any reason for that ?

Those types were generated with the testing. Should I remove them from here and add to testing PR?

**Comment by ayeshLK** [comment]
> Those types were generated with the testing. Should I remove them from here and add to testing PR?

+1 to that.

**Review by ayeshLK — APPROVED** [review]
LGTM

**Pull request merged**